internal/distributed/proxy/httpserver/handler_v1.go

```diff
@@ -0,0 +1,535 @@
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: "insert fail", HTTPReturnError: response.Status.Reason})
+	} else {
+		switch response.IDs.GetIdField().(type) {
+		case *schemapb.IDs_IntId:
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: gin.H{"insertCount": response.InsertCnt, "insertIds": response.IDs.IdField.(*schemapb.IDs_IntId).IntId.Data}})
+		case *schemapb.IDs_StrId:
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: gin.H{"insertCount": response.InsertCnt, "insertIds": response.IDs.IdField.(*schemapb.IDs_StrId).StrId.Data}})
+		default:
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "ids' type neither int or string"})
+		}
+	}
+}
+
+func (h *Handlers) search(c *gin.Context) {
+	httpReq := SearchReq{
+		DbName: DefaultDbName,
+		Limit:  100,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+
+	searchParams := []*commonpb.KeyValuePair{
+		{Key: common.TopKKey, Value: strconv.FormatInt(int64(httpReq.Limit), 10)},
+		{Key: ParamRoundDecimal, Value: "-1"},
+		{Key: ParamOffset, Value: strconv.FormatInt(int64(httpReq.Offset), 10)},
+	}
+	req := milvuspb.SearchRequest{
+		CollectionName:     httpReq.CollectionName,
+		Dsl:                httpReq.Filter,
+		PlaceholderGroup:   vector2PlaceholderGroupBytes(httpReq.Vector),
+		DslType:            commonpb.DslType_BoolExprV1,
+		OutputFields:       httpReq.OutputFields,
+		SearchParams:       searchParams,
+		GuaranteeTimestamp: BoundedTimestamp,
+		Nq:                 int64(1),
+	}
+	username, _ := c.Get(ContextUsername)
+	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
+	if authErr != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusUnauthorized, HTTPReturnMessage: authErr.Error()})
+		return
+	}
+	response, err := h.proxy.Search(c, &req)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "search fail", HTTPReturnError: err.Error()})
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: "search fail", HTTPReturnError: response.Status.Reason})
+	} else {
+		outputData, err := buildQueryResp(response.Results.TopK, httpReq.OutputFields, response.Results.FieldsData, response.Results.Scores)
```

Comment:
The second parameter of "buildQueryResp" should not be directly derived from "http.Req.OutputField"; rather, it should be calculated based on "response.Results".